### ISSUE TITLE
Polling fails on hub restart with TypeError: object Future can't be used in 'await' expression

### ISSUE THREAD

**Issue by ondave** [jupyterhub/kubespawner#466]
On hub restart, poll for existing servers fails with:

```
Traceback (most recent call last):
      File "/usr/local/lib/python3.8/dist-packages/jupyterhub/app.py", line 2032, in check_spawner
        status = await spawner.poll()
      File "/usr/local/lib/python3.8/dist-packages/kubespawner/spawner.py", line 1610, in poll
        await self.pod_reflector.first_load_future
    TypeError: object Future can't be used in 'await' expression
```

I am no expert on the various async libraries, but think this may be because we are using a asyncio await with a concurrent.futures Future?

jupyterhub-1.3.0
jupyterhub-kubespawner-0.14.1
tornado-6.1

**Comment by mriedem** [comment]
Could the fix for this be backported to a 0.14.2 release? We upgraded from 0.13.0 to 0.14.1 yesterday via z2jh 0.10.6 and are hitting this as well (with a lot of pods so it's killing the hub logs on restart).

I can try upgrading to 0.15.0 but worry about picking up more changes since I'm trying to stick as close as possible to the z2jh 0.10.6 stack.

**Comment by mriedem** [comment]
Actually I'll give 0.15.0 a shot, looking at the changelog it doesn't look like it will affect me. The minor version bump scared me for a second but it looks like that's due to the gid change which I don't think will impact us.
